filebeat/inputsource/udp/server_test.go

```diff
@@ -0,0 +1,52 @@
+package udp
+
+import (
+	"net"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestReceiveEventFromUDP(t *testing.T) {
+	tests := []struct {
+		name     string
+		message  []byte
+		expected []byte
+	}{
+		{
+			name:     "Sending a message under the MaxMessageSize limit",
+			message:  []byte("Hello world"),
+			expected: []byte("Hello world"),
+		},
+		{
+			name:     "Sending a message over the MaxMessageSize limit will truncate the message",
+			message:  []byte("Hello world not so nice"),
+			expected: []byte("Hello world not so n"),
+		},
+	}
+
+	ch := make(chan []byte)
+	host := "127.0.0.1:10000"
+	config := &Config{Host: host, MaxMessageSize: 20}
+	fn := func(data []byte) {
+		ch <- data
+	}
+	s := New(config, fn)
+	go func() {
+		s.Run()
+	}()
+	defer s.Stop()
+
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			conn, err := net.Dial("udp", host)
+			if !assert.NoError(t, err) {
+				return
+			}
+			defer conn.Close()
+			conn.Write(test.message)
+			message := <-ch
+			assert.Equal(t, test.expected, message)
+		})
+	}
+}
```

Comment:
We only had python test..

Reply:
It's really great to have these tests in.